### PR TITLE
Feature/download GitHub project

### PR DESCRIPTION
**Pull request by VynnykV**

### CODE REVIEW

File: frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    submit() {
+        this.loading = true;
+        let gitClone: GitClone;
+        if (this.githubRepoSection) {
+            let repository = this.form.value.repository;
+            gitClone = {
+                title: repository.name,
+                url: repository.url,
+                isPublic: this.form.value.isPublic
+            } as GitClone;
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts:97]: Could you move this to a separate method?

File: frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        }
+        else {
+            let repositoryUrl = this.form.value.repositoryUrl;
+            gitClone = {
+                title: this.getNameFromUrl(repositoryUrl),
+                url: repositoryUrl,
+                isPublic: this.form.value.isPublic
+            } as GitClone;
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts:105]: Could you move this to a separate method?

File: frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        return merge(debouncedText$, this.focus$, clicksWithClosedPopup$).pipe(
+            map(term => (term === '' ? this.repositories
+                : this.repositories.filter(r => r.name.toLowerCase().indexOf(term.toLowerCase()) > -1)).slice(0, 25))
+        );
+    }
+
+    submit() {
+        this.loading = true;
+        let gitClone: GitClone;
+        if (this.githubRepoSection) {
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts:91]: Please use ternary operator here

File: frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        this.githubRepoSection = false;
+
+        if (this.form.controls['repository']) {
+            this.form.removeControl('repository');
+        }
+
+        this.form.addControl('repositoryUrl', new FormControl('',
+            [
+                Validators.required,
+                Validators.pattern(`https:\/\/github\.com\/[A-Za-z0-9-]+\/[A-Za-z0-9_.-]+`)
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/import-github-project-dialog/import-github-project-dialog.component.ts:160]: Please move this to our file with regexes

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Please add screenshots =)

**Comment by VynnykV** [comment]
Screenshot_1
Screenshot_2
Screenshot_3
Screenshot_4

**Review by tatianahutii — APPROVED** [review]

**Comment by tatianahutii** [comment]
LGTM

**Pull request merged**